File: m3project/src/main/java/sg/edu/ntu/m3project/m3project/controller/TicketController.java
package sg.edu.ntu.m3project.m3project.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import sg.edu.ntu.m3project.m3project.helper.NewTicket;
import sg.edu.ntu.m3project.m3project.service.TicketService;

@CrossOrigin
@RestController
@RequestMapping("/tickets")
public class TicketController {

        @Autowired
        TicketService ticketService;

        // Get all Tickets
        @GetMapping("/all")
        public ResponseEntity<?> findAll(@RequestHeader(value = "user-id") Integer userId) {
                return ticketService.findAll(userId);
        }

        // Get Tickets by User ID
        @GetMapping
        public ResponseEntity<?> findAllById(@RequestHeader(value = "user-id") Integer userId) {
                return ticketService.findAllById(userId);
        }

        // Create Tickets
        @PostMapping
        public ResponseEntity<?> createTicket(
                        @RequestHeader(value = "user-id") int userId,
                        @RequestBody List<NewTicket> newTickets) {
                return ticketService.add(userId, newTickets);
        }

        // Change Seat
        @PutMapping("/{ticket-id}")
        public ResponseEntity<?> changeSeat(
                        @RequestHeader(value = "user-id") int userId,
                        @PathVariable(value = "ticket-id") int ticketId,
                        @RequestParam String seat) {
                return ticketService.changeSeat(userId, ticketId, seat);
        }

        // Delete ticket
        @PutMapping("/{ticket-id}/delete")
        public ResponseEntity<?> deleteTicket(
                        @RequestHeader(value = "user-id") int userId,
                        @PathVariable(value = "ticket-id") int ticketId) {
                return ticketService.delete(userId, ticketId);
        }

}

File: m3project/src/main/java/sg/edu/ntu/m3project/m3project/controller/TicketController.kt
package sg.edu.ntu.m3project.m3project.controller;

import java.util.*;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.apache.catalina.connector.Response

import sg.edu.ntu.m3project.m3project.helper.NewTicket;
import sg.edu.ntu.m3project.m3project.service.TicketService;

@CrossOrigin
@RestController
@RequestMapping("/tickets")
public class TicketController {

    @Autowired
    var ticketService = TicketService()

    @GetMapping
    fun findAllById(
        @RequestHeader(value = "user_id", required = false) userId : Int?): ResponseEntity<*> {
                return ticketService.find(userId)
        }

    @PostMapping
    fun createTickets(
        @RequestHeader(value = "user_id") userId : Int,
        @RequestBody newTickets : List<NewTicket> ): ResponseEntity<*> {
                return ticketService.add(userId, newTickets)
        }


    @PutMapping("/{ticket_id}")
    fun changeSeat(
        @RequestHeader(value = "user_id") userId : Int,
        @PathVariable ticket_id : Int,
        @RequestBody selectedSeatId : String) : ResponseEntity<*> {
                return ticketService.changeSeat(userId, ticket_id, selectedSeatId)
        }

    @PostMapping("/{ticket_id}/delete")
    fun deleteTicket(
        @RequestHeader(value = "user_id") userId : Int,
        @PathVariable ticket_id : Int) : ResponseEntity<*> {
                return ticketService.delete(userId, ticket_id)
        }
}
